### ISSUE TITLE
Introduce "voicing" as a link relationship 

### ISSUE THREAD

**Issue by murata2makoto** [w3c/epub-specs#1248]
I propose to introduce a link relationship value "voicing" in [Link Relationships](https://w3c.github.io/publ-epub-revision/epub32/spec/epub-packages.html#sec-link-rel).

Use case:  Given a collection of EPUB publications, a blind user would like to choose one of them.  When T2S for the underlying natural language is good, T2S can be used to render the titles of all publications.  But when it is not good enough, we need a mechanism to specify an audio file as the title of each publication. 

Name:	voicing

Description: Indicates that the referenced audio file provides an aural representation of the expression or resource (typically, the title or creator) specified by the refines attribute.

The media type of the audio file is identified in the media-type attribute [Packages] when this keyword is specified.

Cardinality:	Zero or more

Extends: Only applies to the EPUB Publication or collection. MAY be used only when the refines attribute is present.

Example:	`<link refines="#title" rel="voicing" media-type="audio/mpeg" href="title.mp3" />
`

**Comment by dauwhe** [comment]
Are there reading systems that plan to implement this? Is this a common web design pattern?

**Comment by murata2makoto** [comment]
Yes, implementors of three DAISY readers in Japan agree that this is a useful addition.  I'm quite sure that it will be implemented.

**Comment by laudrain** [comment]
@murata2makoto have DAISY readers in Japan implemented EPUB3 ?

**Comment by murata2makoto** [comment]
@laudrain 
Yes, they have done so.

**Comment by llemeurfr** [comment]
@murata2makoto are you speaking about Shinano Kenshi? other device makers?

**Comment by murata2makoto** [comment]
Shinano is one of the three involved in this discussion.   It was Shinano who first pointed out that this feature is missing.

**Comment by mattgarrish** [comment]
> Extends: Only applies to the EPUB Publication or collection. MAY be used only when the refines attribute is present.

I don't believe this adaptation reads quite right. Most link definitions exclude the use of refines because their terms only apply to the publication/collection. You need to make it a MUST to have refines if you're going to provide voicings for specific metadata fields, so maybe:

> Extends: All properties. The refines attribute MUST be present when used.

Also, I believe you copied the wrong example here:

> Example:	`<link refines="#title" rel="voicing" media-type="audio/mpeg" href="title.mp3" />`

You want `href` to reference an `item` in the manifest, correct?

**Comment by laudrain** [comment]
@murata2makoto good news ! I had a call with Shinano on March 7th 2018 about EPUB3 implementation and they seemed to be reluctant at that time.
Are these 3 Japanese companies communicating on their new DAISY readers devices?

**Comment by mikeMaher** [comment]
Hello,

 

We had some people on our development team that were following this.

 

For some reason, they are now forwarding to me.  I tried to track who at scribenet.com is getting these, but cannot figure it out.

 

Can someone tell me the scribenet.com address that you have on your list?

 

I very much would like to discontinue receiving all of these notifications.

 

Thanks,

 

David Alan Rech

President/CEO

Scribe Inc.

 <http://www.scribenet.com> www.scribenet.com

 

main telephone: 215.336.5094

direct telephone: 215.336.5095

facsimile: 215.336.5092

 

3758 SW 30th Avenue, Fort Lauderdale, FL 33312

 

842 South Second Street, Philadelphia, PA 19147

**Comment by mattgarrish** [comment]
Click unwatch at the top of any of the repository pages if you don't want any messages. You're the only one who can control what repositories you're watching.

**Comment by mikeMaher** [comment]
I already did it for Mike and I am still receiving notifications.

There is nothing there for my account (drech@scribenet.com <mailto:drech@scribenet.com> ).

Are there any other scribenet.com accounts on your list?

 

Thanks,

 

David Alan Rech

President/CEO

Scribe Inc.

 <http://www.scribenet.com> www.scribenet.com

 

main telephone: 215.336.5094

direct telephone: 215.336.5095

facsimile: 215.336.5092

 

3758 SW 30th Avenue, Fort Lauderdale, FL 33312

 

842 South Second Street, Philadelphia, PA 19147

**Comment by mattgarrish** [comment]
One other thought would be to take a look on the right-hand side of this issue and see if GitHub has you subscribed specifically to it.

Otherwise, you might try contacting GitHub support to see if they have any ideas. This group can't help with platform issues.

**Comment by murata2makoto** [comment]
I am a committee member for the migration from DAISY 2 to EPUB 3 for accessible textbooks.  Shinano, Saccessnet, and Cypac are members.  Japanese Society for Rehabilitation of Persons with Disabilities (JSRPD) is the secretariat.  MEXT (Ministry of Education, Culture, Sports, Science and Technology) is an observer.  Hiroshi Kawamura is the chair.

DAISY 2.02 lacks ruby.   This is already a good reason for the migration.   Japan has abused DAISY 2.02 by introducing ruby, and we have seen interoperability problems.  I also know that Shinano has adopted picture and section elements heavily in their latest editor.  Two other new features are planned for EPUB3 accessible textbooks.

It was Shinano who first pointed out the lack of a mechanism for specifying voice title files.  Saccessnet seconded immediately.  I am just requesting the consensus of a meeting of the committee mentioned above.

Adoption of EPUB 3 is planned to happen next April.  But volunteers for creating accessible textbooks are not yet convinced if the authoring environment is mature.  But even if the adoption is delayed, it will eventually happen and we are working on it aggressively.

**Comment by murata2makoto** [comment]
@mattgarrish 

> Extends: All properties. The refines attribute MUST be present when used.

Agreed, but replace "when used" by "when this value is used".


> Example:	<link refines="#title" rel="voicing" media-type="audio/mpeg" href="title.mp3" />
> You want href to reference an item in the manifest, correct?

I find that examples for [alternate](https://w3c.github.io/publ-epub-revision/epub32/spec/epub-packages.html#sec-alternate) and other values directly reference file names.  Which is better?

**Comment by RachelComerford** [comment]
Thanks @murata2makoto - I appreciate the reasoning behind the request. My co-chair is currently on vacation and won't be able to attend this weeks CG and so I am going to hold off on discussing whether adding additional features to 3.2 is appropriate at this stage (especially considering its additional impact on EPUBCheck). We'll add this as an agenda item for 4/11/19.

@dauwhe FYI for your agenda that week

**Comment by mattgarrish** [comment]
> I find that examples for [alternate](https://w3c.github.io/publ-epub-revision/epub32/spec/epub-packages.html#sec-alternate) and other values directly reference file names. Which is better?

Well, I was forgetting about this:

> Linked resources are not Publication Resources and MUST NOT be listed in the manifest.

A resource is only supposed to be listed if it's also used in a content document.

I suppose that renders any discussion kind of moot. It's probably for the best, since it's less indirection to get what you need to specify the file directly. `link` also allows the media type to be specified, which is probably the most useful additional bit of information to know since linked resources don't fall under core media type rules. (I don't know if that's a good thing or a bad thing in this case.)

**Comment by murata2makoto** [comment]
If I am not mistaken, the only required change is adding a line to the
definition of LINKREL_PROPERTIES in java/com/adobe/epubcheck/vocab/PackageVocabs.java.

**Comment by dauwhe** [comment]
An alternative would be to use existing methods of extending EPUB metadata with the `meta` element

```xml
<meta refines="#title" property="foo:voicing">title.mp3</meta>
```

**Comment by murata2makoto** [comment]
@dauwhe True.  But do we have a plan for future versions of EPUB 3?  Now that we have removed external vocabularies, every addition requires a new version of EPUB 3.

**Comment by dauwhe** [comment]
Closed via #1257
